### ISSUE TITLE
AudioStreamPlayer.get_playback_position() updating very slowly

### ISSUE THREAD

**Issue by wingedadventurer** [godotengine/godot#44871]
**Godot version:**
3.2.3.stable

**OS/device including version:**
Windows 10
Audio driver: WASAPI

**Issue description:**
Calling `AudioStreamPlayer.get_playback_position()` doesn't return accurate enough result. It doesn't update frequently enough.

I compared it with custom variable that I increase by `delta` every regular frame or physics frame, and the stream playback is updating very slowly.

Check the minimal reproduction project below. It displays 2 progress bars, one for stream playback position, other for custom playback position variable. You'll see stream playback being very jittery compared to custom playback. Using wav or ogg format doesn't make a difference.

I'm making a rhythm game and it's pretty essential to be able to get precise playback time any frame.

**Minimal reproduction project:**
[get_playback_position.zip](https://github.com/godotengine/godot/files/5760494/get_playback_position.zip)

**Comment by wingedadventurer** [comment]
I've done some source code digging (not too deep, though).

Inside `Audio_stream_player.cpp`:
```
float AudioStreamPlayer::get_playback_position() {
	...
	return stream_playback->get_playback_position();
	...
}
```

`stream_playback` is an `AudioStreamPlayback` class, declared inside `audio_stream.h`. `get_playback_position()` is virtual.

This is inheritance chain:

```
AudioStreamPlayback
	AudioStreamPlaybackRandomPitch
	AudioStreamPlaybackSample
	AudioStreamPlaybackResampled
		AudioStreamPlaybackMP3
		AudioStreamPlaybackOGGVorbis
		AudioStreamPlaybackMicrophone
		AudioStreamGeneratorPlayback
```

`AudioStreamPlaybackSample` looks related to WAV files, and `AudioStreamPlaybackOGGVorbis` to OGG files.

`audio_stream_sample.cpp` (WAV)
```
float AudioStreamPlaybackSample::get_playback_position() const {
	return float(offset >> MIX_FRAC_BITS) / base->mix_rate;
}
```

`audio_stream_ogg_vorbis.cpp` (OGG)
```
float AudioStreamPlaybackOGGVorbis::get_playback_position() const {
	return float(frames_mixed) / vorbis_stream->sample_rate;
}
```

**Comment by wingedadventurer** [comment]
I failed to check the docs properly. The explanation for the delay, and solution for it, is explained in the documentation:
https://docs.godotengine.org/en/stable/tutorials/audio/sync_with_audio.html

TLDR:

Audio is mixed in chunks (depending on mix rate), and get_playback_position() will update only when next chunk is loaded. This is the way audio playback works.

To get the time between each mix, you can use `AudioServer.get_time_since_last_mix()`. So to get almost exact playback position, you do `get_playback_position() + AudioServer.get_time_since_last_mix()`.

This is not super precise either (there's slight jitter that needs some tweaking), but it's good enough for my case.
